spec/haskell/src/SEL4/Object/SchedContext.lhs

```diff
@@ -664,10 +664,9 @@ This module uses the C preprocessor to select a target architecture.
 >         threadSet (\tcb -> tcb { tcbSchedContext = Nothing }) from
 >         cur <- getCurThread
 >         action <- getSchedulerAction
->         case action of
->             SwitchToThread candidate -> when (candidate == from || from == cur) rescheduleRequired
->             _ -> when (from == cur) rescheduleRequired
->     setSchedContext scPtr (sc { scTCB = Just tcbPtr })
+>         when (from == cur || action == SwitchToThread from)
+>             rescheduleRequired
+>     updateSchedContext scPtr (\sc -> sc { scTCB = Just tcbPtr })
```

Comment:
It was kind of unexpected to see a haskell update in a commit that says "prove ..." with no further elaboration. It looks like this is a rephrase to make the proof easier, but it still left me scratching my head for a bit as to what's going on.

Reply:
Yes, this is just a rephrase to make the proof easier. This is what's happening the vast majority of the time when I update the Haskell, which was written seemingly without any thought as to the Simpl that will be produced. I can add a bit to the commit message saying that it is just to make the proof easier

Reply:
Yes please. I think it's important knowledge that this is only a rephrase and not a bug fix or update to match the C.